stan/math/prim/fun/log_mix.hpp

```diff
@@ -157,38 +91,19 @@ return_type_t<T_theta, T_lam> log_mix(const T_theta& theta,
   check_finite(function, "theta", theta);
   check_consistent_sizes(function, "theta", theta, "lambda", lambda);
 
-  scalar_seq_view<T_theta> theta_vec(theta);
-  T_partials_vec theta_dbl(N);
-  for (int n = 0; n < N; ++n) {
-    theta_dbl[n] = value_of(theta_vec[n]);
-  }
-
-  scalar_seq_view<T_lam> lam_vec(lambda);
-  T_partials_vec lam_dbl(N);
-  for (int n = 0; n < N; ++n) {
-    lam_dbl[n] = value_of(lam_vec[n]);
-  }
+  const auto& theta_dbl = value_of(as_column_vector_or_scalar(theta));
+  const auto& lam_dbl = value_of(as_column_vector_or_scalar(lambda));
 
   T_partials_return logp = log_sum_exp((log(theta_dbl) + lam_dbl).eval());
-
-  T_partials_vec theta_deriv(N);
-  theta_deriv.array() = (lam_dbl.array() - logp).exp();
-
-  T_partials_vec lam_deriv = theta_deriv.cwiseProduct(theta_dbl);
+  T_partials_vec theta_deriv = (lam_dbl.array() - logp).exp();
 
   operands_and_partials<T_theta, T_lam> ops_partials(theta, lambda);
-  if (!is_constant_all<T_theta>::value) {
-    for (int n = 0; n < N; ++n) {
-      ops_partials.edge1_.partials_[n] = theta_deriv[n];
-    }
-  }
-
   if (!is_constant_all<T_lam>::value) {
-    for (int n = 0; n < N; ++n) {
-      ops_partials.edge2_.partials_[n] = lam_deriv[n];
-    }
+    ops_partials.edge2_.partials_ = theta_deriv.cwiseProduct(theta_dbl);
```

Comment:
I'm not sure how this compiles when ```T_lam``` is an ```std::vector<T>```, wouldn't there be an ```operator=``` error for trying to assign an ```Eigen``` vector to an ```std::vector```? Or is the ```partials_``` member always an Eigen vector? I haven't worked with ```operands_and_partials``` in a while

Reply:
I checked it and `partials_` is always an Eigen vector.